### PR TITLE
Remove unused includes

### PR DESCRIPTION
**Pull request by ccawley2011**
Common system headers are included from `src/common.h`, and duplicate or unused includes have been removed.

### CODE REVIEW

File: src/period.c [hunk 1]
```diff
@@ -19,9 +19,6 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-#ifndef _GNU_SOURCE
-#define _GNU_SOURCE
```
Comment (sezero) [inline src/period.c:23]: This was added by commit 50d111a859e0e995bcbab57cb4fcfee3ab5ebf43 because round() requires _GNU_SOURCE on some old systems.

Comment (ccawley2011) [reply src/period.c:23]: OK, fixed.

**Review by sezero — APPROVED** [review]
The `lite` side requires the following patch too:
```diff
diff --git a/lite/src/format.c b/lite/src/format.c
index 41dc7d0..829c48e 100644
--- a/lite/src/format.c
+++ b/lite/src/format.c
@@ -20,11 +20,6 @@
  * THE SOFTWARE.
  */
 
-#include <stdlib.h>
-#include <string.h>
-#ifndef LIBXMP_CORE_PLAYER
-#include "loaders/prowizard/prowiz.h"
-#endif
 #include "format.h"
 
 extern const struct format_loader libxmp_loader_xm;
```
Ok from me with that.  @AliceLR: Any objections?

**Comment by AliceLR** [comment]
I did some test builds and everything seems to work fine on my end (MSYS2: normal, -DDEBUG, and --disable-prowizard --disable-depackers; Fedora: normal; Android NDK via MegaZeux to make sure the load_helpers.c change didn't break). No complaints here.

**Pull request merged**